lib/workers/package-file/dep-type.js

```diff
@@ -104,6 +106,14 @@ function getDepConfig(depTypeConfig, dep) {
           }
         }
       }
+      if (matchVersion && packageNames.includes(depConfig.depName)) {
```

Comment:
Why check `packageNames` ?

Reply:
To not apply the matchVersion rule to all dependencies but just the ones in packageNames.

Reply:
That's not the logic I wanted, I will fix/rewrite it for you